test/jdk/sun/net/www/http/KeepAliveStream/KeepAliveStreamCloseWithWrongContentLength.java

```diff
@@ -23,91 +23,162 @@
 
 /*
  * @test
- * @bug 4533243
- * @summary Closing a keep alive stream gives NullPointerException
+ * @bug 4533243 8263364
+ * @summary Closing a keep alive stream should not give NullPointerException and should accept a connection from  a
+ *          client only from this test
  * @library /test/lib
  * @run main/othervm/timeout=30 KeepAliveStreamCloseWithWrongContentLength
  */
 
 import java.net.*;
 import java.io.*;
 import jdk.test.lib.net.URIBuilder;
+import java.nio.ByteBuffer;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
 
 public class KeepAliveStreamCloseWithWrongContentLength {
 
-    static class XServer extends Thread {
-        ServerSocket srv;
-        Socket s;
-        InputStream is;
-        OutputStream os;
+    private final static String path = "/KeepAliveStreamCloseWithWrongContentLength";
+    private final static String getRequest1stLine = "GET %s".formatted(path);
 
-        XServer (ServerSocket s) {
-            srv = s;
+    static class XServer extends Thread implements AutoCloseable {
+
+        final ServerSocket serverSocket;
+        volatile Socket clientSocket;
+
+        XServer (InetAddress address) throws IOException {
+            ServerSocketChannel serverSocketChannel = ServerSocketChannel.open();
+            ServerSocket serversocket = serverSocketChannel.socket();
+            serversocket.bind(new InetSocketAddress(address, 0));
+            this.serverSocket = serversocket;
+        }
+
+        public int getLocalPort() {
+            return serverSocket.getLocalPort();
         }
 
         public void run() {
+
             try {
-                s = srv.accept ();
-                // read HTTP request from client
-                InputStream is = s.getInputStream();
-                // read the first ten bytes
-                for (int i=0; i<10; i++) {
-                    is.read();
+                ByteArrayOutputStream clientBytes;
+                clientSocket = null;
+
+                // in a concurrent test environment it can happen that other rouge clients connect to this server
```

Comment:
I guess clients are rogue and not rouge ;-)
Maybe we should clean that up in the mainline next time we revisit this test...